gpt4all-backend/llamamodel.cpp

```diff
@@ -432,7 +432,18 @@ std::vector<LLModel::Token> LLamaModel::tokenize(PromptContext &ctx, const std::
 
 std::string LLamaModel::tokenToString(Token id) const
 {
-    return llama_token_to_piece(d_ptr->ctx, id);
+    std::vector<char> result(8, 0);
```

Comment:
There are two versions of llama_token_to_piece. The removed call is to [the `static` one](https://github.com/ggerganov/llama.cpp/blob/e1b40ac3b94824d761b5e26ea1bc5692706029d9/llama.cpp#L1649-L1662) internal to the source file `llama.cpp`, which we had to patch in order to expose it via the C API (which is especially a hack since it returns std::string and breaks the unit tests written in C). The new call is to the [standard one](https://github.com/ggerganov/llama.cpp/blob/e1b40ac3b94824d761b5e26ea1bc5692706029d9/llama.h#L857-L867) actually exposed in the upstream llama.h, since I removed the patch when I rebased.